cmd/influxd/launcher/cmd.go

```diff
@@ -70,24 +70,34 @@ func cmdRunE(ctx context.Context, o *InfluxdOpts) func() error {
 	return func() error {
 		fluxinit.FluxInit()
 
-		// exit with SIGINT and SIGTERM
-		ctx = signals.WithStandardSignals(ctx)
-
 		l := NewLauncher()
-		if err := l.run(ctx, o); err != nil {
+		if err := l.runUntilDone(ctx, o); err != nil {
 			return err
 		}
-		<-ctx.Done()
-
-		// Attempt clean shutdown.
-		ctx, cancel := context.WithTimeout(ctx, 2*time.Second)
```

Comment:
This is the bug:
* At this point, `ctx` is completed (because we got past the `<-ctx.Done()` call above)
* `context.WithTImeout` returns an already-completed context if the 1st arg is completed

Reply:
I think you need a new context here, because the parent forces the new child context closed, as well. WithTimeout() calls WithDeadline() which has the following comment:

/ WithDeadline returns a copy of the parent context with the deadline adjusted
// to be no later than d. If the parent's deadline is already earlier than d,
// WithDeadline(parent, d) is semantically equivalent to parent. The returned
// context's Done channel is closed when the deadline expires, when the returned
// cancel function is called, or when the parent context's Done channel is
// closed, whichever happens first.